CoffeeChat/Controller/EditInterestViewController.swift

```diff
@@ -0,0 +1,195 @@
+            : moreInterests[indexPath.section - yourInterests.count]
+    }
+
+}
+
+// MARK: - UITableViewDelegate
+extension EditInterestViewController: UITableViewDelegate {
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let interest = getInterestFromIndexPath(indexPath)
+        if indexPath.section < yourInterests.count { // From Top Section
+            yourInterests.removeAll(where: { $0.name == interest.name })
+            moreInterests.append(interest)
+            moreInterests.sort(by: {$0.name < $1.name })
+        } else { // Bottom Section
+            moreInterests.removeAll(where: { $0.name == interest.name })
+            yourInterests.append(interest)
+            yourInterests.sort(by: {$0.name < $1.name })
+        }
+        tableView.reloadData()
+    }
+}
+
+// MARK: - UITableViewDataSource
+extension EditInterestViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier:
+            OnboardingTableViewCell.reuseIdentifier, for: indexPath) as?
+            OnboardingTableViewCell else { return UITableViewCell() }
+
+        let data = getInterestFromIndexPath(indexPath)
+        cell.configure(with: data)
+        cell.selectionChangesAppearence(false)
+        cell.contentView.backgroundColor = indexPath.section < yourInterests.count ? .pearGreen : .white
+        return cell
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        let cellSpacing: CGFloat = 12
+        let headerSpacing: CGFloat = 88
+        return [0, yourInterests.count].contains(section) ? headerSpacing : cellSpacing
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView: UIView
+
+        if section == yourInterests.count { // Lower Section
+            headerView = InterestHeaderView()
+            if let headerView = headerView as? InterestHeaderView {
+                headerView.configure(with: "More Interests", info: "Tap to select")
+                headerView.backgroundColor = .none
+            }
+        } else if section == 0 { // Upper Section
+            headerView = InterestHeaderView()
+            if let headerView = headerView as? InterestHeaderView {
+                headerView.configure(with: "Your Interests", info: "Tap to deselect")
+                headerView.backgroundColor = .none
+            }
+        } else { // Filler Space
+            headerView = UIView()
+        }
+        return headerView
+    }
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return yourInterests.count + moreInterests.count
+    }
```

Comment:
what's this function for?

Reply:
atm since each cell is a section, the numberOfSections = number of total cells